packages/visualizations/src/components/Map/utils.js

```diff
@@ -53,65 +52,94 @@ export const mapKeyToColor = (values, colorScale) => {
     const scale = chroma.scale([colorMin, colorMax]).domain([min, max]);
 
     const mapping = {};
-    
+
     values.forEach(v => {
         mapping[v.x] = scale(v.y).hex();
     });
 
     return mapping;
 }
 
-// This is a bound that can only be extended, and will take the value of the first bound that extends it
-const VOID_BOUNDS = [[
-    180, 
+// This is a default bound that will be extended
+const VOID_BOUNDS = [
+    180,
     90,
-], [
-    -180, 
+    -180,
     -90,
-]];
+];
+
+function computeBboxFromCoords(coordsPath, bbox) {
+    return coordsPath.reduce((current, coords) => {
+        return [
+            Math.min(coords[0], current[0]),
+            Math.min(coords[1], current[1]),
+            Math.max(coords[0], current[2]),
+            Math.max(coords[1], current[3]),
+        ]
+    }, bbox);
+}
+
+// Because features come from tiled vector data, feature geometries may be split
+// or duplicated across tile boundaries. As a result, features may appear
+// multiple times in query results.
+// To prevent looping on duplicated features in computeMaxZoomFromGeoJsonFeatures we merge
```

Comment:
Can we use `new Set(features)` to dedupe? If there are no differences in features at all… or `array.filter` else maybe?

Reply:
👍  I changed the function to avoid looping through arrays, it should be better now !